payment_paypal/indico_payment_paypal/plugin.py

```diff
@@ -73,6 +75,93 @@ def adjust_payment_form_data(self, data):
+                            choice['price_info']  = billable_choices[0]['price_info']
+                            choice['itemNum']     = ii
+                            itemized_data.append(choice)
+
+                elif regdata.field_data.field.input_type == 'single_choice':
+                    billable_choices = [x for x in versioned_data['choices'] if x['id'] in regdata.data and x['is_billable']]
+                    if billable_choices:
+                        ii = ii + 1
+                        choice = {}
+                        choice['title']       = fieldTitle + ' - ' + unversioned_data['captions'][billable_choices[0]['id']]
+                        choice['price']       = format_currency(billable_choices[0]['price'], '', u'#0.00', locale='en_US') 
+                        choice['quantity']    = regdata.data[billable_choices[0]['id']]
+                        choice['price_total'] = regdata.price
+                        choice['price_info']  = billable_choices[0]['price_info']
+                        choice['itemNum']     = ii
+                        itemized_data.append(choice)
+                 
+                elif regdata.field_data.field.input_type == 'multi_choice':
+                    billable_choices = [x for x in versioned_data['choices'] if x['id'] in regdata.data and x['is_billable']]
+                    if billable_choices:
+                        for bc in billable_choices:                            
+                            ii = ii + 1
+                            choice = {}
+                            choice['title']       = fieldTitle + ' - ' + unversioned_data['captions'][bc['id']]
+                            choice['price']       = format_currency(bc['price'], '', u'#0.00', locale='en_US') 
+                            choice['quantity']    = regdata.data[bc['id']]
+                            choice['price_total'] = bc['price'] * regdata.data[bc['id']]
+                            choice['price_info']  = bc['price_info']
+                            choice['itemNum']     = ii
+                            itemized_data.append(choice)
+
+                else: 
+                    if regdata.price:                        
+                        ii = ii + 1
+                        choice = {}
+                        choice['title']       = sectionTitle + ' - ' + fieldTitle
+                        choice['price']       = format_currency(versioned_data['price'], '', u'#0.00', locale='en_US') 
+                        if regdata.field_data.field.input_type == 'number':
+                            choice['quantity'] = regdata.data
+                        else:
+                            choice['quantity'] = 1
+                        choice['price_total'] = regdata.price
+                        choice['price_info']  = versioned_data['price_info']
+                        choice['itemNum']     = ii
+                        itemized_data.append(choice)
```

Comment:
If you look at this block of code, you'll notice that you are repeating quite often the same lines (or at least paraphrasing them). This is normally a sign that behaviour can be extracted to a function. You could have a `create_choice_data(...)` function that takes as parameters the `billable_choice`, the item number and a couple more things, and then generates the dict you need. BTW, you can do that like this:
```python
choice = {
    'title': '{} - {}'.format(section_title, field_title),
    # ...
}
```
No need to do the creation and assignment in separate steps.